compiler/src/main/scala/org/scalajs/core/compiler/GenJSCode.scala

```diff
@@ -3092,9 +3324,55 @@ abstract class GenJSCode extends plugins.PluginComponent
       val receiver = genExpr(receiver0)
       val args = genPrimitiveJSArgs(sym, args0)
 
+      genJSCallGeneric(sym, receiver, args, isStat)
+    }
+
+    private def genJSSuperCall(tree: Apply, isStat: Boolean): js.Tree = {
+      implicit val pos = tree.pos
+      val Apply(fun @ Select(sup @ Super(_, mix), _), args) = tree
+      val sym = fun.symbol
+
+      val genReceiver = genThis()(sup.pos)
+      lazy val genScalaArgs = genActualArgs(sym, args)
+      lazy val genJSArgs = genPrimitiveJSArgs(sym, args)
+
+      if (sym.owner == ObjectClass) {
+        // Normal call anyway
+        assert(!sym.isClassConstructor,
+            "Trying to call the super constructor of Object in a " +
+            s"Scala.js-defined JS class at $pos")
+        genApplyMethod(genReceiver, sym, genScalaArgs)
+      } else if (sym.isClassConstructor) {
+        // Implementation restriction
+        if (genJSArgs.exists(_.isInstanceOf[js.JSSpread])) {
+          reporter.error(pos,
+              "Implementation restriction: cannot call the super " +
+              "constructor of a Scala.js-defined JS class with :_*")
+        }
+
+        js.JSSuperConstructorCall(genJSArgs)
+      } else if (isScalaJSDefinedJSClass(sym.owner) && !isExposed(sym)) {
+        // Reroute to the static method
+        genApplyJSClassMethod(genReceiver, sym, genScalaArgs)
+      } else {
+        genJSCallGeneric(sym, genReceiver, genJSArgs, isStat,
+            superIn = Some(currentClassSym))
```

Comment:
Also, doesn't this hit the most simple `genCall` case only in `genJSCallGeneric`? And you could just replace this with:

```scala
js.JSBracketSuperCall(
 jstpe.ClassType(encodeClassFullName(superInSym)),
 receiver, methodName, args)
```